driver/src/test/functional/com/mongodb/reactivestreams/client/gridfs/GridFSPublisherSpecification.groovy

```diff
@@ -126,7 +127,36 @@ class GridFSPublisherSpecification extends FunctionalSpecification {
         def data = runAndCollect(gridFSBucket.&downloadToPublisher, fileId)
 
         then:
-        concatByteBuffers(data) == concatByteBuffers([ByteBuffer.wrap(contentBytes), ByteBuffer.wrap(contentBytes)])
+        concatByteBuffers(data) == contentBytes
+    }
+
+    def 'should respect the outer subscription request amount'() {
+        given:
+        def contentBytes = multiChunkString.getBytes()
+        def options = new GridFSUploadOptions().chunkSizeBytes(contentBytes.length)
+
+        when:
+        def publisher = gridFSBucket.uploadFromPublisher('myFile',
+        createPublisher(ByteBuffer.wrap(contentBytes), ByteBuffer.wrap(contentBytes),
+                ByteBuffer.wrap(contentBytes)), options)
+
+        def subscriber = new ObservableSubscriber()
+        publisher.subscribe(subscriber)
+        def fileId = subscriber.await(1, 60, SECONDS).getReceived().get(0)
+
+        then:
+        run(filesCollection.&countDocuments) == 1
+        run(chunksCollection.&countDocuments) == 3
+
+        when:
+        subscriber = new ObservableSubscriber()
+        publisher = gridFSBucket.downloadToPublisher(fileId as ObjectId).bufferSizeBytes(contentBytes.length * 3)
```

Comment:
Same for the download - requesting 1 value here should equate to all the chunks from the db.

Reply:
Where are we requesting 1?  In the ObservableSubscriber?  

Why should requesting 1 equal to all chunks?  The request isn't for a certain number of bytes?